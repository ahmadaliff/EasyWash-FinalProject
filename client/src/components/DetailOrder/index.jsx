import Proptypes from 'prop-types';
import { Dialog, DialogContent, DialogTitle, IconButton } from '@mui/material';
import { Close, DryCleaning } from '@mui/icons-material';
import { FormattedMessage, FormattedNumber, injectIntl } from 'react-intl';
import MapRouting from '@components/MapRouting';

import classes from '@components/DetailOrder/style.module.scss';

const EditProfile = ({ order, open, handleClose }) => (
  <Dialog open={open} onClose={handleClose} fullWidth>
    <DialogTitle fontSize="larger">
      <FormattedMessage id="app_detail_order" />
    </DialogTitle>
    <IconButton
      aria-label="close"
      onClick={handleClose}
      sx={{
        position: 'absolute',
        right: 8,
        top: 8,
      }}
    >
      <Close />
    </IconButton>
    <DialogContent>
      <div>
        <div className={classes.field}>
          <p className={classes.label}>
            <FormattedMessage id="app_order_id" />
          </p>
          <p className={classes.text}>{order?.id}</p>
        </div>
        <div className={classes.field}>
          <p className={classes.label}>
            <FormattedMessage id="app_total_price" />
          </p>
          <p className={classes.text}>
            {
              // eslint-disable-next-line react/style-prop-object
              <FormattedNumber value={order?.totalPrice} style="currency" currency="IDR" minimumFractionDigits={0} />
            }
          </p>
        </div>
        <div className={classes.field}>
          <p className={classes.label}>Status</p>
          <p className={classes.text}>
            <FormattedMessage id={order?.status} />
          </p>
        </div>
        {order?.Services?.map((service, key) => (
          <div className={classes.service} key={key}>
            <DryCleaning />
            <div className={classes.serviceInfoWrap}>
              <hr />
              <div className={classes.serviceInfo}>
                <p>
                  {service?.name} x {service?.ServicesOrdered?.quantity}
                </p>
                <p>
                  <FormattedNumber
                    // eslint-disable-next-line no-unsafe-optional-chaining
                    value={service?.price * service?.ServicesOrdered?.quantity}
                    // eslint-disable-next-line react/style-prop-object
                    style="currency"
                    currency="IDR"
                    minimumFractionDigits={0}
                  />
                </p>
              </div>
              <hr />
            </div>
          </div>
        ))}
        <div className={classes.field}>
          <p className={classes.label}>
            <FormattedMessage id="app_location" />
          </p>
        </div>
        <MapRouting
          laundryPoint={order && JSON.parse(order?.Services[0]?.Merchant?.location)}
          userPoint={order && JSON.parse(order?.location)}
        />
      </div>
    </DialogContent>
  </Dialog>
);

EditProfile.propTypes = {
  order: Proptypes.object,
  open: Proptypes.bool,
  handleClose: Proptypes.func,
};

export default injectIntl(EditProfile);
