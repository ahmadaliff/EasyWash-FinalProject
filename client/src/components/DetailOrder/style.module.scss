@import '@styles/base';

.field {
  @include directionX;
  align-items: center;
  font-size: 0.75rem;
  font-weight: 500;
  .label {
    float: left;
    font-weight: 700;
    width: 10rem;
  }
  .text {
    width: 100%;
  }
}
