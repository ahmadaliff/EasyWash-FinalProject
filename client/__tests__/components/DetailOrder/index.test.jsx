import { render, fireEvent, screen } from '@utils/testHelper';
import DetailOrder from '@components/DetailOrder';

jest.mock('@components/MapRouting', () => jest.fn(() => <div>mapleafletMock</div>));

jest.mock('react-dom/server', () => ({
  renderToStaticMarkup: jest.fn(),
}));

let isOpen = true;

const mockProps = {
  open: isOpen,
  order: {
    id: 1,
    totalPrice: 20000,
    status: 'app_pending',
    Services: [
      {
        name: 'service',
        ServicesOrdered: {
          quantity: 2,
        },
        price: 10000,
        Merchant: { location: '{"lat": -6.224934545453234, "lng": 106.86266439802276 }' },
      },
    ],
    location: '{"lat": -6.224934545453234, "lng": 106.86266439802276 }',
  },
  handleClose: () => {
    isOpen = false;
  },
};

let wrapper;

beforeEach(() => {
  wrapper = render(<DetailOrder {...mockProps} />);
});

describe('Detail Order', () => {
  test('Should render correctly', () => {
    const { getByTestId } = wrapper;
    expect(getByTestId('dialog-detail')).toBeInTheDocument();
  });
  test('Should call navigate when button clicked', () => {
    const { getByTestId } = wrapper;
    const button = getByTestId('button-close');
    fireEvent.click(button);
    expect(isOpen).toBe(false);
  });
  test('Should be render text', () => {
    expect(screen.getByText('Detail pesanan')).toBeInTheDocument();
    expect(screen.getByText('Id Pesanan')).toBeInTheDocument();
    expect(screen.getByText('Harga Total')).toBeInTheDocument();
    expect(screen.getByText('Status')).toBeInTheDocument();
    expect(screen.getByText('Ditunda')).toBeInTheDocument();
  });
  test('Should match with snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
