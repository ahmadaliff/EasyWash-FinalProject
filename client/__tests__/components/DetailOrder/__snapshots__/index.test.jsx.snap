// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail Order Should match with snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 1024px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    />
    <div
      class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
      data-testid="dialog-detail"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
        role="presentation"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        tabindex="-1"
      >
        <div
          aria-labelledby=":r3:"
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiDialog-paperFullWidth css-tlc64q-MuiPaper-root-MuiDialog-paper"
          role="dialog"
        >
          <h2
            class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-1xi8exm-MuiTypography-root-MuiDialogTitle-root"
            id=":r3:"
          >
            <b>
              Detail pesanan
            </b>
          </h2>
          <button
            aria-label="close"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-qsfli3-MuiButtonBase-root-MuiIconButton-root"
            data-testid="button-close"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CloseIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <div
            class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
          >
            <div>
              <div
                class="field"
              >
                <p
                  class="label"
                >
                  Id Pesanan
                </p>
                <p
                  class="text"
                >
                  1
                </p>
              </div>
              <div
                class="field"
              >
                <p
                  class="label"
                >
                  Harga Total
                </p>
                <p
                  class="text"
                >
                  RpÂ 20.000
                </p>
              </div>
              <div
                class="field"
              >
                <p
                  class="label"
                >
                  Status
                </p>
                <p
                  class="text"
                >
                  Ditunda
                </p>
              </div>
              <div
                class="service"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="DryCleaningIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19.56 11.36 13 8.44V7c0-.55-.45-1-1-1s-1-.45-1-1 .45-1 1-1 1 .45 1 1h2c0-1.84-1.66-3.3-3.56-2.95-1.18.22-2.15 1.17-2.38 2.35-.3 1.56.6 2.94 1.94 3.42v.63l-6.56 2.92c-.88.38-1.44 1.25-1.44 2.2v.01C3 14.92 4.08 16 5.42 16H7v6h10v-6h1.58c1.34 0 2.42-1.08 2.42-2.42v-.01c0-.95-.56-1.82-1.44-2.21zM18.58 14H17v-1H7v1H5.42c-.23 0-.42-.19-.42-.43 0-.17.1-.32.25-.38l6.75-3 6.75 3c.15.07.25.22.25.39 0 .23-.19.42-.42.42z"
                  />
                </svg>
                <div
                  class="serviceInfoWrap"
                >
                  <hr />
                  <div
                    class="serviceInfo"
                  >
                    <div
                      class="name"
                    >
                      <p>
                        service
                         x 
                        2
                      </p>
                      <p
                        class="type"
                      >
                        Tipe  
                        Satuan
                      </p>
                    </div>
                    <p>
                      RpNaN
                    </p>
                  </div>
                  <hr />
                </div>
              </div>
              <div
                class="field"
              >
                <p
                  class="label"
                >
                  Lokasi
                </p>
              </div>
              <div>
                mapleafletMock
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
